lib/containers/container_images.pm

```diff
@@ -240,4 +240,22 @@ sub upload_3rd_party_images_logs {
     }
 }
 
+sub test_systemd_install {
+    my %args = @_;
+    my $image = $args{image};
+    my $runtime = $args{runtime};
+
+    die 'Argument $image not provided!' unless $image;
+    die 'Argument $runtime not provided!' unless $runtime;
+
+    my ($image_version, $image_sp, $image_id) = get_os_release("$runtime run --entrypoint '' $image");
+    # TW and starting with SLE 15-SP4/Leap15.4 systemd's dependency with udev has been dropped
+    if ($image_id eq 'opensuse-tumbleweed' ||
+        ($image_id eq 'opensuse-leap' && check_version('>=15.4', "$image_version.$image_sp", qr/\d{2}\.\d/)) ||
+        ($image_id eq 'sles' && check_version('>=15-SP4', "$image_version-SP$image_sp", qr/\d{2}-sp\d/))) {
+        assert_script_run "$runtime run $image zypper al udev";
+        assert_script_run "$runtime run $image zypper -n in systemd";
```

Comment:
should we also fail if `rpm -q --requires systemd | grep udev !=0`  in the container?

Reply:
The trick here is that `zypper` will not proceed if `systemd` will need to pull `udev` as the lock prevents it to be installed. Maybe to make it completely clear, there might be a check that ensures `udev` is not pre-installed.

Reply:
but I think you are creating a new fresh container in each command... so the lock won't do anything for the `zypper -n in` command.  If you wanna keep this way, we need to combine both commands in the same `docker run` command, something like:
```
assert_script_run "$runtime run $image /bin/bash -c \"zypper -n in systemd\"";
```

Reply:
So, for 15-SP3: (as expected)

```
# podman run -ti registry.suse.de/suse/sle-15-sp3/update/cr/totest/images/suse/sle15:15.3  /bin/bash -c "zypper al udev && zypper -n in systemd"
Specified lock has been successfully added.
Refreshing service 'container-suseconnect-zypp'.
Problem retrieving the repository index file for service 'container-suseconnect-zypp':
[container-suseconnect-zypp|file:/usr/lib/zypp/plugins/services/container-suseconnect-zypp] 
Warning: Skipping service 'container-suseconnect-zypp' because of the above error.
Building repository 'SLE_BCI' cache ..........................................................................................................................................[done]
Loading repository data...
Reading installed packages...
Resolving package dependencies...

Problem: the to be installed systemd-246.16-7.28.1.x86_64 requires 'udev = 246.16-7.28.1', but this requirement cannot be provided
  not installable providers: udev-246.16-7.28.1.x86_64[SLE_BCI]
 Solution 1: remove lock to allow installation of udev-246.16-7.28.1.x86_64[SLE_BCI]
 Solution 2: do not install systemd-246.16-7.28.1.x86_64
 Solution 3: break systemd-246.16-7.28.1.x86_64 by ignoring some of its dependencies

Choose from above solutions by number or cancel [1/2/3/c/d/?] (c): c
```

Fails, as the expected dependency fails, but the exact same command for `registry.suse.de/suse/sle-15-sp4/update/cr/totest/images/suse/sle15:15.4` works.


However, if you split it in 2 calls, it works for 15-SP3 because the containers are created for each `run` cmd.

Reply:
Yup, you are right, changed! Thanks!